### PR TITLE
CI: Add release based containers

### PR DESCRIPTION
**Pull request by syeopite**
- Changes the current master based container to use "master" tag instead of "latest" tag
- Adds a new workflow to build a container on each new release which has  the "latest" tag, and a tag based on the current released version.

Once this process is done, it should allow us to quickly merge PRs which passed the code-review but not the testing stage. Instead, the testing-stage can be pushed to when each release occurs.

### CODE REVIEW

**Comment by iBicha** [comment]
> Changes the current master based container to use "master" tag instead of "latest" tag

`stable` might be more commonly used/understood. Or even `production`.
Very good initiative!

**Comment by syeopite** [comment]
The release containers are tagged as "latest" as to ensure a smooth transition for instances. 

As testing is being moved to per-release with this PR, instances should be using the release containers instead of the container built off master as to ensure instance stability.

**Comment by iBicha** [comment]
> The release containers are tagged as "latest" as to ensure a smooth transition for instances.
> 
> As testing is being moved to per-release with this PR, instances should be using the release containers instead of the container built off master as to ensure instance stability.

Fair enough, I take my comment back

**Comment by SamantazFox** [comment]
Could you rename the two files to something in the odds of `container-release` and `container-nightly` (or whatever you see fit)? The two names are too similar, that's confusing!

File: .github/workflows/build-stable-container.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: quay.io/invidious/invidious
+          tags: |
+            type=semvar,pattern={{version}}
```
Comment (SamantazFox) [inline .github/workflows/build-stable-container.yml:50]: Small typo
```suggestion
            type=semver,pattern={{version}}
```
Reference: https://github.com/docker/metadata-action?tab=readme-ov-file#typesemver

File: .github/workflows/build-stable-container.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Crystal
+        uses: crystal-lang/install-crystal@v1.8.2
+        with:
+          crystal: 1.12.2
```
Comment (SamantazFox) [inline .github/workflows/build-stable-container.yml:19]: Commenting here, but can you bump the other file to the same version?

File: .github/workflows/build-stable-container.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      - name: Docker meta
+        id: meta-arm64
+        uses: docker/metadata-action@v5
+        with:
+          images: quay.io/invidious/invidious
+          flavor: |
+            suffix=-arm64
+          tags: |
+            type=semvar,pattern={{version}}
```
Comment (SamantazFox) [inline .github/workflows/build-stable-container.yml:75]:
```suggestion
            type=semver,pattern={{version}}
```

**Review by SamantazFox — CHANGES_REQUESTED** [review]

**Comment by SamantazFox** [comment]
Marking as ready as the only option to trigger that is to merge the code and publish a new tagged release.

**Review by unixfox — APPROVED** [review]

**Pull request merged**